lib/jekyll/jekyll-feed.rb

```diff
@@ -32,7 +32,7 @@ def source_path
 
     def feed_content
       feed = PageWithoutAFile.new(@site, File.dirname(__FILE__), "", path)
-      feed.content = File.read(source_path).gsub(/\s+([{<])/, '\1')
+      feed.content = File.read(source_path).gsub(/(?<!\")\s+([<{])/, '\1')
```

Comment:
Would it have also worked to put the space on the other side of the `hreflang`, i.e. at the beginning (right after `{% if %}`) instead of at the end?